ibis/pandas/core.py

```diff
@@ -168,6 +168,13 @@ def execute_without_scope(
     params = {k.op() if hasattr(k, 'op') else k: v for k, v in params.items()}
 
     new_scope = toolz.merge(scope, data_scope, params, factory=factory)
+
+    # pre_execute
```

Comment:
this is ``execute_without_scope`` we need to do this here as well.

Reply:
This already happens in `execute_with_scope` though. Why do we need to call it here?

Reply:
because this was getting called *first*, I am using ``execute(t)``, does that call ``execute_without_scope``?

I had a ``data_preload`` defined as well. These shouldn't be mutually exclusive. (and conceputally easier to put all of the data access in a single place as well).

Reply:
`execute_with_scope` calls `execute_without_scope`

maybe `data_preload` can go away if we add `client` to the dispatch. for now you should probably just define `pre_execute` since you want to do those optimizations.